unit-tests/sensor/frames/test-filter.cpp

```diff
@@ -0,0 +1,116 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+// We have our own main
+#define CATCH_CONFIG_MAIN
+#define CATCH_CONFIG_RUNNER
+
+#include "../../catch.h"
+
+#include <easylogging++.h>
+#ifdef BUILD_SHARED_LIBS
+// With static linkage, ELPP is initialized by librealsense, so doing it here will
+// create errors. When we're using the shared .so/.dll, the two are separate and we have
+// to initialize ours if we want to use the APIs!
+INITIALIZE_EASYLOGGINGPP
+#endif
+
+#include <librealsense2/rs.hpp>  // Include RealSense Cross Platform API
+#include <chrono>
+#include <atomic>
+
+using namespace std::chrono;
+
+
+// Test description:
+// This test verify that if the user ask for depth stream only, the incoming frames will be depth frames only.
+// L515 device has a frame filter that filters IR frames when depth only is requested
+// Note: * L515 specific test
+//       * Test will pass if no L515 device is connected
+//       * Test will be performed on the first L515 device detected.
+TEST_CASE( "Sensor", "[frames-filter]" )
+{
+    try
+    {
+        static const int FRAMES_TO_CHECK = 100;
+        static const int TEST_TIMEOUT_MS = 10000;
+        rs2::context ctx;
+        rs2::config cfg;
+        rs2::device_list devices;
+
+        devices = ctx.query_devices();
+        bool l515_found = false;
+        std::atomic<int>  depth_frames_count{ 0 };
+        std::atomic<int>  non_depth_frames_count{ 0 };
+
+        for (auto &&dev : devices)
+        {
+            bool dev_is_l515 = dev.supports(RS2_CAMERA_INFO_NAME) ?
+                std::string(dev.get_info(RS2_CAMERA_INFO_NAME)).find("L515") != std::string::npos : false;
+
+            if (dev_is_l515) // First L515 device found
+            {
+                l515_found = true;
+
+                auto depth = devices[0].first<rs2::depth_sensor>();
+
+                auto stream_profiles = depth.get_stream_profiles();
+                // Find depth stream
+                auto depth_stream = std::find_if(
+                    stream_profiles.begin(),
+                    stream_profiles.end(),
+                    [](rs2::stream_profile sp) { return sp.stream_type() == RS2_STREAM_DEPTH; });
+
+                if (depth_stream != stream_profiles.end())
```

Comment:
If this doesn't pass, don't we want to fail the test?

Reply:
We do, added